slack/slack.go

```diff
@@ -0,0 +1,73 @@
+package slack
+
+import (
+	"encoding/json"
+	"errors"
+
+	resty "gopkg.in/resty.v1"
+)
+
+var (
+	errorBadResponse = "API call gave an other-than-200 result"
+	errorEmptyText   = "Refusing to send an empty message, consider calling slack.SetText() first"
+)
+
+// API provides an interface to interact with the Slack Webhook API
+type API struct {
```

Comment:
This name is very generic, you may add another descriptive name

Reply:
full type name (with `slack`) will be `slack.API` - not sure if still bad - have no preference.

Reply:
Yes, I prefer to keep it is API considering the package name is `slack`.

Reply:
You may commit go standard convention for services which is Doer, so I prefer something like `Provider`

Reply:
done!